src/components/views/CoursesView.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+  <div>
+    <div class="row">
+      <div
+        class="col-12 col-md-6 col-lg-4"
+        v-for="course in courses"
+        :key="course.id"
+      >
+        <div class="card-wrapper">
+          <course-card :course="course" :enrollments="enrollments" />
+        </div>
+      </div>
+    </div>
+    <loading v-if="isLoading" />
+    <div
+      class="row mt-lg-5 justify-content-md-center align-items-center"
+      v-else-if="courses.length === 0"
+    >
+      <span class="col col-lg-4 font-weight-bold"
+        >Sorry, no matching courses..</span
+      >
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import CourseCard from "@/components/course/CourseCard";
+import { tokenMixin } from "@/components/mixins/tokenMixin";
+import { enrollmentsMixin } from "@/components/mixins/courseMixin";
+import Loading from "@/components/Loading";
+
+export default {
+  name: "CoursesView",
+  components: {
+    Loading,
+    CourseCard,
+  },
+  mixins: [tokenMixin, enrollmentsMixin],
+
+  data() {
+    return {
+      courses: [],
+      selectedTopic: null,
+      searchString: null,
+      filteredCourses: null,
+      isLoading: true,
+    };
+  },
+
+  methods: {
+    filterCourses() {
+      if (this.selectedTopic !== null) {
+        let filtered = [];
```

Comment:
Da sa nahradit za:
```
const filtered = this.courses.filter(c => c.topic === this.selectedTopic)
```
Realne by to mal ale riesit BE podla mna - cize by sa mal zavolat novy request na backend.